src/overlays/actors/ovl_En_Daiku2/z_en_daiku2.c

```diff
@@ -22,56 +53,525 @@ const ActorInit En_Daiku2_InitVars = {
+    Actor_SetHeight(&this->actor, 65.0f);
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    Math_ApproachF(&this->unk_260, this->unk_264, 0.3f, 2.0f);
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1D);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    func_80BE7600(this, globalCtx);
+}
 
-extern ColliderCylinderInit D_80BE7900;
-#endif
+void EnDaiku2_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnDaiku2* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE61D0.asm")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/EnDaiku2_Init.asm")
+    func_8012C28C(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/EnDaiku2_Destroy.asm")
+    if (limbIndex == 14) {
+        Matrix_Scale(this->unk_260, this->unk_260, this->unk_260, 1);
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, D_06009638);
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE6408.asm")
+    if (limbIndex == 15) {
+        gSPDisplayList(POLY_OPA_DISP++, D_0600A390);
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE64C0.asm")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE65B4.asm")
+void EnDaiku2_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku2* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE66E4.asm")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE6B40.asm")
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gDPSetEnvColor(POLY_OPA_DISP++, 245, 155, 0, 255);
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount, NULL,
+                     EnDaiku2_PostLimbDraw, &this->actor);
+    func_80BE7718(this, globalCtx);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE6BC0.asm")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE6CFC.asm")
+void func_80BE7504(EnDaiku2* this, Vec3f* arg1, Vec3f* arg2, Vec3f* arg3, f32 arg4, s16 arg5) {
+    s16 i;
+    EnDaiku2Particle* particle = &this->particles[0];
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Daiku2_0x80BE61D0/func_80BE6D40.asm")
+    for (i = 0; i < ARRAY_COUNT(this->particles); i++, particle++) {
+        if (particle->enabled == false) {
```

Comment:
Again, feel free to ignore, but I think this is more readable 

```suggestion
        if (!particle->enabled) {
```